bigquery/bigquery_storage_quickstart/main.go

```diff
@@ -197,9 +199,13 @@ func valueFromTypeMap(field interface{}) interface{} {
 // successfully transmitted.
 func processStream(ctx context.Context, client *bqStorage.BigQueryStorageClient, st *bqStoragepb.Stream, ch chan<- *bqStoragepb.AvroRows) error {
 	var offset int64
-	streamRetry := 3
+
+	// Streams may be long-running.  Rather than using a global retry for the
+	// stream, implement a retry that resets once progress is made.
+	retryLimit := 3
 
 	for {
+		retries := retryLimit
```

Comment:
Optional: Intuitively, `retries` seems like it should start at 0 to me. Makes sense once you look below and check it's being decremented, but while you're changing this, consider making it start a 0.

Reply:
Swapped.